File: letter_usage/solutions/letter_usage.js
var word = "My name is Fred";

var letters = {};

for(var i = 0; i < word.length; i++) {
  console.log(word[i]);
  console.log(letters[word[i]]);
  if (letters[word[i]] === undefined) {
    letters[word[i]] = 1;
  }
  else {
    var amount = letters[word[i]];
    letters[word[i]] = amount + 1;
  }
}
console.log(letters);

File: letter_usage/solutions/letter_usage.java
import java.util.*;
class Letter_Usage {
  public static void main(String[] args) {

    String word = "My name is Fred";
    char[] words = word.toCharArray();
    Map<Character, Integer> letters = new HashMap<>();
    
    
    for(char letter : words){
      if (letters.containsKey(letter)){
        int amount = letters.get(letter);
        letters.put(letter, (amount + 1));
      }
      else {
        letters.put(letter, 1);
      }
      
    }
    
    System.out.println(letters);
    
  }
  

  
}